gap/oper.gi

```diff
@@ -275,7 +302,32 @@ end);
 
 InstallMethod(DigraphRemoveEdges, "for an immutable digraph and a list",
 [IsImmutableDigraph, IsList],
-{D, edges} -> MakeImmutable(DigraphRemoveEdges(DigraphMutableCopy(D), edges)));
+function(D, edges)
+   local DMutableRemoved;
+   DMutableRemoved := DigraphRemoveEdges(DigraphMutableCopy(D), edges);
+
+   # If any edges have been removed
+   if DigraphNrEdges(D) <> DigraphNrEdges(DMutableRemoved) then
```

Comment:
Again, could we somehow check this before copying, to avoid the extra work if it's not necessary?

I think you could check all the inputs with `IsDigraphEdge` rather than comparing numbers after a copy.